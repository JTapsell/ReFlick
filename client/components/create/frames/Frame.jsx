import React from 'react'
import { setFrame } from '../../../actions'
import { connect } from 'react-redux'

function Frame (props) {
  const { setFrame, frame, update } = props
  return (
    <div className="frame" onClick={() => {
      setFrame(frame)
      update()
    }}>
      <p className="frame-number">{props.frame}</p>
      <br/>
    </div>
  )
}

function mapsStateToDispatch (dispatch) {
  return {
    setFrame: (frame) => dispatch(setFrame(frame))
  }
}

export default connect(null, mapsStateToDispatch)(Frame)
