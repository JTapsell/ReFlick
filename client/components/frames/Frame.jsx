import React from 'react'
import { setFrame } from '../../actions'
import { connect } from 'react-redux'

function Frame (props) {
  const { setFrame, frame } = props
  return (
    <div className="frame" onClick={() => setFrame(frame)}>
      <p>{props.frame}</p>
      <br/>
    </div>
  )
}

function mapsStateToDispatch (dispatch) {
  return {
    setFrame: (frame) => dispatch(setFrame(frame))
  }
}

export default connect(null, mapsStateToDispatch)(Frame)
