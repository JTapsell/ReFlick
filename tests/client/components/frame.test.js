import Frame from '../../../client/components/frames/Frame'
import { shallow } from 'enzyme'
import React from 'react'

test('test runner is working', () => {
  expect(true).toBeTruthy()
})

let wrapper

beforeEach(() => {
  wrapper = shallow(<Frame/>)
})
describe('<Frame is running', () => {
  it('Renders correct', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('Renders one <p>', () => {
    expect(wrapper.find('p')).toHaveLength(1)
  })
})
