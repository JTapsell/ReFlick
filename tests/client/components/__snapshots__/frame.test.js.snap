// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Frame is running Renders correct 1`] = `ShallowWrapper {}`;
